drivers/display/ls0xx.c

```diff
@@ -0,0 +1,375 @@
+#define LS0XX_PANEL_HEIGHT  DT_INST_PROP(0, height)
+
+#define LS0XX_PIXELS_PER_BYTE  8U
+/* Adding 2 for the line number and dummy byte
+ * line_buf format for each row.
+ * +-------------------+-------------------+----------------+
+ * | line num (8 bits) | data (WIDTH bits) | dummy (8 bits) |
+ * +-------------------+-------------------+----------------+
+ */
+#define LS0XX_BYTES_PER_LINE  ((LS0XX_PANEL_WIDTH / LS0XX_PIXELS_PER_BYTE) + 2)
+
+#define LS0XX_BIT_WRITECMD    0x01
+#define LS0XX_BIT_VCOM        0x02
+#define LS0XX_BIT_CLEAR       0x04
+
+static uint8_t line_buf[LS0XX_BYTES_PER_LINE];
+
+static struct spi_buf tx_buf;
+static struct spi_buf_set buf_set = { .buffers = &tx_buf, .count = 1 };
+static uint8_t extcomin_bit;
+
+struct ls0xx_data {
+#if DT_INST_NODE_HAS_PROP(0, disp_en_gpios)
+	const struct device *disp_dev;
+#endif
+#if DT_INST_NODE_HAS_PROP(0, extcomin_gpios)
+	const struct device *extcomin_dev;
+#endif
+	const struct device *spi_dev;
+	struct spi_cs_control cs_ctrl;
+	struct spi_config spi_config;
+};
+
+#if defined(CONFIG_LS0XX_VCOM_DRIVER) && \
+	DT_INST_NODE_HAS_PROP(0, extcomin_gpios)
+/* Driver will handle VCOM toggling */
+static void ls0xx_vcom_toggle(void *a, void *b, void *c)
+{
+	struct ls0xx_data *driver = (struct ls0xx_data *)a;
+
+	while (1) {
+		gpio_pin_toggle(driver->extcomin_dev,
+				DT_INST_GPIO_PIN(0, extcomin_gpios));
+		k_usleep(3);
+		gpio_pin_toggle(driver->extcomin_dev,
+				DT_INST_GPIO_PIN(0, extcomin_gpios));
+		k_msleep(1000 / DT_INST_PROP(0, extcomin_frequency));
+	}
+}
+
+K_THREAD_STACK_DEFINE(vcom_toggle_stack, 256);
+struct k_thread vcom_toggle_thread;
+#endif
+
+static int ls0xx_blanking_off(const struct device *dev)
+{
+#if DT_INST_NODE_HAS_PROP(0, disp_en_gpios)
+	struct ls0xx_data *driver = dev->data;
+
+	return gpio_pin_set(driver->disp_dev,
+			    DT_INST_GPIO_PIN(0, disp_en_gpios), 1);
+#else
+	LOG_WRN("Unsupported");
+	return -ENOTSUP;
+#endif
+}
+
+static int ls0xx_blanking_on(const struct device *dev)
+{
+#if DT_INST_NODE_HAS_PROP(0, disp_en_gpios)
+	struct ls0xx_data *driver = dev->data;
+
+	return gpio_pin_set(driver->disp_dev,
+			    DT_INST_GPIO_PIN(0, disp_en_gpios), 0);
+#else
+	LOG_WRN("Unsupported");
+	return -ENOTSUP;
+#endif
+}
+
+static int ls0xx_clear(const struct device *dev)
+{
+	struct ls0xx_data *driver = dev->data;
+	int err;
+	uint8_t clear_cmd[2] = { LS0XX_BIT_CLEAR | extcomin_bit, 0 };
+
+	tx_buf.len = sizeof(uint16_t);
+	tx_buf.buf = &clear_cmd;
+	err = spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+	spi_release(driver->spi_dev, &driver->spi_config);
+
+	return err;
+}
+
+static int ls0xx_update_display(const struct device *dev,
+				uint16_t start_line,
+				uint16_t num_lines,
+				const uint8_t *buf)
+{
+	struct ls0xx_data *driver = dev->data;
+	int err;
+	uint8_t write_cmd;
+	uint16_t max_line;
+
+	LOG_DBG("Lines %d to %d", start_line, start_line + num_lines - 1);
+
+#if defined(CONFIG_LS0XX_VCOM_DRIVER) && \
```

Comment:
@jfischer-no 
The only concern i have regarding this is what if the user requires the GPIO pin. Toggling via SPI would free up a GPIO for the user.

Reply:
Well, I would recommend to choose a different display or use an I²C I/O Expander.

Reply:
@jfischer-no 
Alright. Also, what do you think about using a pwm instead of gpio toggle?

Reply:
I see the advantage of the _option_ to use a PWM pin, but I think requiring it isn't a great choice, as it will limit the applications in which you can use the display (in any setting where you are out of pwm pins, or just making pin selection a little less convenient).

Right now a user can choose to not define CONFIG_LS0XX_VCOM_DRIVER and set up a pwm pin to do that manually (if I understand correctly) without requiring all (extcomin based) applications to use a pwm resource.

Reply:
> @jfischer-no
> Alright. Also, what do you think about using a pwm instead of gpio toggle?

@rgujju That could be added later, let's make it ready to be merged with minimal but clear functionality.